### PR TITLE
chore: reduce spellcheck ignore words range

### PR DESCRIPTION
**Pull request by septs**

### CODE REVIEW

**Review by Jack-Works — DISMISSED** [review]

**Comment by UncleBill** [comment]
Should not be listed in change log, use `chore` instead of `feat`

File: packages/icons/plugins/Dhedge.tsx [hunk 1]
```diff
@@ -1,6 +1,8 @@
 import { createIcon } from '../utils'
 import type { SvgIcon } from '@mui/material'
 
+// cspell:ignore dhedge
```
Comment (UncleBill) [inline packages/icons/plugins/Dhedge.tsx:4]: add to `web3-tokens`

File: cspell.json [hunk 2]
```diff
@@ -1,396 +1,225 @@
 {
     "$schema": "https://github.com/streetsidesoftware/cspell/raw/main/cspell.schema.json",
-    "dictionaries": ["typescript", "node", "html", "css", "fonts"],
+    "dictionaries": ["typescript", "node", "html", "css", "fonts", "web3-tokens"],
+    "dictionaryDefinitions": [
+        {
+            "name": "web3-tokens",
+            "path": "packages/web3-constants/tokens.txt"
+        }
```
Comment (UncleBill) [inline cspell.json:8]: add one more dictionary `web3` to manage `eth` `btc` `dhedge`, `opensea`, `uniswap`, `poap` and so on

File: packages/icons/plugins/index.tsx [hunk 3]
```diff
@@ -1,3 +1,4 @@
+// cspell:ignore dhedge
```
Comment (UncleBill) [inline packages/icons/plugins/index.tsx:1]: list in `web3` dictionary

File: packages/injected-script/main/locationChange.ts [hunk 4]
```diff
@@ -5,6 +5,7 @@ function setupChromium() {
     window.history.pushState = new no_xray_Proxy(history.pushState, {
         apply(target, thisArg, params: any) {
             const val = apply(target, thisArg, params)
+            // cspell:ignore pushstate
```
Comment (UncleBill) [inline packages/injected-script/main/locationChange.ts:8]: should be listed in dictionary that like `web` or `javascript`?

File: packages/injected-script/main/locationChange.ts [hunk 5]
```diff
@@ -13,6 +14,7 @@ function setupChromium() {
     window.history.replaceState = new no_xray_Proxy(history.replaceState, {
         apply(target, thisArg, params: any) {
             const val = apply(target, thisArg, params)
+            // cspell:ignore replacestate
```
Comment (UncleBill) [inline packages/injected-script/main/locationChange.ts:17]: ditto

File: packages/injected-script/rollup.config.js [hunk 6]
```diff
@@ -1,5 +1,7 @@
 import sucrase from '@rollup/plugin-sucrase'
 
+// cspell:ignore iife
```
Comment (UncleBill) [inline packages/injected-script/rollup.config.js:3]: quite common usage in JavaScript.

File: packages/mask/src/plugins/External/README.md [hunk 7]
```diff
@@ -1,5 +1,7 @@
 # External plugins
 
+<!-- cspell:ignore everytime -->
```
Comment (UncleBill) [inline packages/mask/src/plugins/External/README.md]: A typo indeed, should be `every time`

File: packages/mask/src/plugins/RedPacket/Worker/database.ts [hunk 8]
```diff
@@ -3,6 +3,8 @@ import type { RedPacketRecord, RedPacketRecordInDatabase, RedPacketNftRecordInDa
 import { RedPacketPluginID } from '../constants'
 import { createPluginDatabase } from '../../../database/Plugin'
 
+// cspell:ignore redpackets
```
Comment (UncleBill) [inline packages/mask/src/plugins/RedPacket/Worker/database.ts:6]: `redpacket` should be treated as a word, which created by us.

File: packages/mask/src/plugins/VCent/apis/index.ts [hunk 9]
```diff
@@ -1,5 +1,7 @@
 import { TWEET_BASE_URL } from '../constants'
 
+// cspell:ignore txid
```
Comment (UncleBill) [inline packages/mask/src/plugins/VCent/apis/index.ts:3]: Let's accept `txid` as a word, quite command usage.

File: packages/mask/src/plugins/Wallet/apis/opensea.ts [hunk 10]
```diff
@@ -2,6 +2,8 @@ import { ChainId } from '@masknet/web3-shared-evm'
 import urlcat from 'urlcat'
 import { OPENSEA_API_KEY } from '../constants'
 
+// cspell:ignore presale
```
Comment (UncleBill) [inline packages/mask/src/plugins/Wallet/apis/opensea.ts:5]: presale, or pre-sale, quite commonly seen in token tokenomics.

File: packages/mask/src/plugins/CryptoartAI/SNSAdaptor/CheckoutDialog.tsx [hunk 11]
```diff
@@ -17,6 +17,8 @@ import { activatedSocialNetworkUI } from '../../../social-network'
 import { isTwitter } from '../../../social-network-adaptor/twitter.com/base'
 import { isFacebook } from '../../../social-network-adaptor/facebook.com/base'
 
+// cspell:ignore webm
```
Comment (UncleBill) [inline packages/mask/src/plugins/CryptoartAI/SNSAdaptor/CheckoutDialog.tsx:20]: should be list in web technology dictionary.

File: packages/mask/src/utils/hooks/index.ts [hunk 12]
```diff
@@ -1,3 +1,4 @@
+// cspell:ignore switcher
```
Comment (UncleBill) [inline packages/mask/src/utils/hooks/index.ts:1]: I think it's quite common usage.

File: packages/mask/src/utils/theme.ts [hunk 13]
```diff
@@ -16,6 +16,8 @@ import { Subscription, useSubscription } from 'use-subscription'
 import produce, { setAutoFreeze } from 'immer'
 import twitterColorSchema from '../social-network-adaptor/twitter.com/customization/twitter-color-schema.json'
 
+// cspell:ignore monofont monospace consolas
```
Comment (UncleBill) [inline packages/mask/src/utils/theme.ts:19]: should be list in web technology dictionary.

File: packages/plugins/Solana/src/apis/shared.ts [hunk 14]
```diff
@@ -2,6 +2,8 @@ import type { ChainId } from '@masknet/web3-shared-solana'
 import { TOKEN_PROGRAM_ID } from '@solana/spl-token'
 import { NETWORK_ENDPOINTS } from '../constants'
 
+// cspell:ignore lamports
```
Comment (UncleBill) [inline packages/plugins/Solana/src/apis/shared.ts:5]: list in `web3 dictionary. `lamport` is like `wei` to `eth`

File: packages/polyfills/builder.mjs [hunk 15]
```diff
@@ -5,6 +5,8 @@ import { readFile, writeFile } from 'fs/promises'
 import { createRequire } from 'module'
 import { createHash } from 'crypto'
 
+// cspell:ignore pnpm
```
Comment (UncleBill) [inline packages/polyfills/builder.mjs:8]: should be list in web technology dictionary.

File: packages/provider-proxy/rollup.config.js [hunk 16]
```diff
@@ -6,6 +6,8 @@ import alias from '@rollup/plugin-alias'
 import dts from 'rollup-plugin-dts'
 import replace from '@rollup/plugin-replace'
 
+// cspell:ignore treeshake
```
Comment (UncleBill) [inline packages/provider-proxy/rollup.config.js:9]: should be list in web technology dictionary.

File: packages/injected-script/main/locationChange.ts (hunk 4 above)
Comment (septs) [reply packages/injected-script/main/locationChange.ts:8]:
```console
$ npx cspell trace 'pushstate' 'replacestate'
No matches found
```

**Review by Jack-Works — APPROVED** [review]

File: packages/injected-script/main/locationChange.ts (hunk 4 above)
Comment (UncleBill) [reply packages/injected-script/main/locationChange.ts:8]: then we can add into our dictionary.

**Pull request closed without merging**